File: utilslib/src/main/java/com/ashlikun/utils/other/RomUtils.java
package com.ashlikun.utils.other;

/**
 * 作者　　: 李坤
 * 创建时间: 2018/8/3　13:42
 * 邮箱　　：496546144@qq.com
 * <p>
 * 功能介绍：手机rom的工具
 */
public class RomUtils {
    public static final String ROM_XIAOMI = "XIAOMI";
    public static final String ROM_HUAWEI = "HUAWEI";
    public static final String ROM_FLYME = "FLYME";
    public static final String ROM_OPPO = "OPPO";
    //锤子
    public static final String ROM_SMARTISAN = "SMARTISAN";
    public static final String ROM_SAMSUNG = "SAMSUNG";
    public static final String ROM_VIVO = "VIVO";
    //360
    public static final String ROM_QIKU = "QIKU";
    public static final String ROM_360 = "360";
    private final static String ZUKZ1 = "zuk z1";
    private final static String ZTEC2016 = "zte c2016";

    public static boolean isXiaomi() {
        return check(ROM_XIAOMI);
    }

    public static boolean isHuawei() {
        return check(ROM_HUAWEI);
    }

    public static boolean isVivo() {
        return check(ROM_VIVO);
    }

    public static boolean isOppo() {
        return check(ROM_OPPO);
    }

    public static boolean isFlyme() {
        return check(ROM_FLYME);
    }

    public static boolean is360() {
        return check(ROM_QIKU) || check(ROM_360);
    }

    public static boolean isSmartisan() {
        return check(ROM_SMARTISAN);
    }

    public static boolean isSamsung() {
        return check(ROM_SAMSUNG);
    }

    /**
     * 判断是否为 ZUK Z1 和 ZTK C2016。
     * 两台设备的系统虽然为 android 6.0，但不支持状态栏icon颜色改变，因此经常需要对它们进行额外判断。
     */
    public static boolean isZUKZ1() {
        final String board = android.os.Build.MODEL;
        return board != null && board.toLowerCase().contains(ZUKZ1);
    }

    public static boolean isZTKC2016() {
        final String board = android.os.Build.MODEL;
        return board != null && board.toLowerCase().contains(ZTEC2016);
    }

    public static boolean isEssentialPhone() {
        return DeviceUtil.getDeviceBrand().contains("essential");
    }

    private static boolean check(String rom) {
        String brand = DeviceUtil.getDeviceBrand();
        String manufacturer = DeviceUtil.getManufacturer();
        if (brand != null && rom.equals(brand.toUpperCase())) {
            return true;
        }
        if (manufacturer != null && rom.equals(manufacturer.toUpperCase())) {
            return true;
        }
        return false;
    }
}

File: utilslib/src/main/java/com/ashlikun/utils/other/RomUtils.kt
package com.ashlikun.utils.other

import android.os.Build
import com.ashlikun.utils.other.DeviceUtil.Companion.deviceBrand
import com.ashlikun.utils.other.DeviceUtil.Companion.manufacturer

/**
 * @author　　: 李坤
 * 创建时间: 2021.12.13 16:38
 * 邮箱　　：496546144@qq.com
 *
 * 功能介绍：手机rom的工具
 */

object RomUtils {
    const val ROM_XIAOMI = "XIAOMI"
    const val ROM_HUAWEI = "HUAWEI"
    const val ROM_FLYME = "FLYME"
    const val ROM_MEIZU = "MEIZU"
    const val ROM_OPPO = "OPPO"
    const val ROM_OnePlus = "OnePlus"

    //锤子
    const val ROM_SMARTISAN = "SMARTISAN"
    const val ROM_SAMSUNG = "SAMSUNG"
    const val ROM_VIVO = "VIVO"

    //360
    const val ROM_QIKU = "QIKU"
    const val ROM_360 = "360"
    const val ZUKZ1 = "zuk z1"
    const val ZTEC2016 = "zte c2016"

    /**
     * 判断是否为 ZUK Z1 和 ZTK C2016。
     * 两台设备的系统虽然为 android 6.0，但不支持状态栏icon颜色改变，因此经常需要对它们进行额外判断。
     */
    val isZUKZ1: Boolean
        get() = Build.MODEL != null && Build.MODEL.contains(ZUKZ1, true)

    val isXiaomi: Boolean
        get() = check(ROM_XIAOMI)
    val isHuawei: Boolean
        get() = check(ROM_HUAWEI)
    val isVivo: Boolean
        get() = check(ROM_VIVO)
    val isOppo: Boolean
        get() = check(ROM_OPPO)
    val isMeizu: Boolean
        get() = check(ROM_FLYME) || check(ROM_MEIZU)
    val isOnePlus: Boolean
        get() = check(ROM_OnePlus)

    fun is360(): Boolean {
        return check(ROM_QIKU) || check(ROM_360)
    }

    val isSmartisan: Boolean
        get() = check(ROM_SMARTISAN)
    val isSamsung: Boolean
        get() = check(ROM_SAMSUNG)
    val isZTKC2016: Boolean
        get() {
            val board = Build.MODEL
            return board != null && board.uppercase().contains(ZTEC2016)
        }
    val isEssentialPhone: Boolean
        get() = deviceBrand.contains("essential")

    private fun check(rom: String): Boolean {
        if (rom.uppercase() == deviceBrand.uppercase()) return true
        if (rom.uppercase() == manufacturer.uppercase()) return true
        return false
    }
}